### DISCUSSION TITLE
Help with False Positive regex

### DISCUSSION THREAD

**Discussion by Serrindipity** [ltex-plus/vscode-ltex-plus#75] [Q&A]
First of all, I wanted to say thanks for this fork. I do all my writing and note-taking in VSCode so this is super helpful.

I'm currently trying to add a rule in `ltex.hiddenFalsePositives.en-US.txt` that hides false positives in Wikilinks (i.e. `[[ text | link ]]` syntax). I've tried the following, to no avail. Anyone have any advice for debugging this kind of regex / what it might look like? Thanks!

```json
{"rule":"MORFOLOGIK_RULE_EN_US","sentence":".*?]]"}
{"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\[\[.*?]]"} // This errors and crashes ltex
{"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\\[\\[.*?]]"}
{"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\\[\\[.*?\\]\\]"}
{"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\\\\[\\\\[.*?\\\\]\\\\]"}
```

Plus, is it possible that the way ltex+ parses sentences ignores / post-processes the brackets?

**Comment by spitzerd** [comment]
Regarding your first question, please try with two backslashes
`{"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\\[\\[.*?]]"}`
You have to escape it twice, because TypeScript/JavaScript/Java uses the backslash symbol as escape symbol and regex as well. If you use just one backslash, not the regex is escaped but the TypeScript/JavaScript/Java code. This causes a crash of the program.

Regarding your second question, I don't know either if this is possible.

**Reply by Serrindipity** [reply]
I've tried it with 2 backslashes, as above:

> ```json
> {"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\\[\\[.*?]]"}
> ```

as well as 
> ```json
> {"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\\[\\[.*?\\]\\]"}
> ```
which doesn't work either.

I've tried each of the lines in the original post separately, but none of them seem to work, hence my second question.

**Comment by spitzerd** [comment]
That's weird. For `[[ text | link ]]`, I get a spelling error as well by default.  Both `{"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\\[\\[.*?]]"}` and `{"rule":"MORFOLOGIK_RULE_EN_US","sentence":"\\[\\[.*?\\]\\]"}` hide the spelling error.

**Comment by Serrindipity** [comment]
Yeah that is really weird. I've tried both of those and I still get errors. I know the file is being read correctly, since when I use **Hide false positive** in the Quick Fix menu it adds to the same file, and hides whatever I've added correctly. Any other suggestions for debugging that I could try?

**Reply by spitzerd** [reply]
What happens if you change the generated file from _Hide false positive in the Quick Fix menu_?

**Reply by Serrindipity** [reply]
Doesn't work, I add the line to the file but it still shows the spelling error.
